### PR TITLE
Add package: hyfetch

### PR DESCRIPTION
**Pull request by EtaoinWu**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

**Notes**. [Hyfetch](https://github.com/hykilpikonna/hyfetch) is a system info display tool (similar to the now deprecated `neofetch`) with customizable pride flag color schemes. The original developer has approved the submission.

### CODE REVIEW

**Comment by EtaoinWu** [comment]
@conda-forge/help-some-language, ready for review!

**Comment by EtaoinWu** [comment]
@conda-forge/help-python, ready for review!

File: recipes/hyfetch/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  entry_points:
+    - hyfetch=hyfetch.main:run
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  host:
+    - python
+    - pip
+    - typing-extensions
```
Comment (BastianZim) [inline recipes/hyfetch/meta.yaml:23]: Is this really required for building? I cannot find it upstream.

**Review by BastianZim — CHANGES_REQUESTED** [review]
Just one small thing.

File: recipes/hyfetch/meta.yaml (hunk 1 above)
Comment (EtaoinWu) [reply recipes/hyfetch/meta.yaml]: Removed with 181f35d. Hyfetch v1.4.10 required it to build, but upstream commit [5519dd1](https://github.com/hykilpikonna/hyfetch/commit/5519dd16a065193e85805c0e66b273fcd067993b) removed such dependency.

**Review by xhochy — APPROVED** [review]

**Pull request merged**